torch/_dynamo/variables/builtin.py

```diff
@@ -1375,6 +1375,11 @@ def _comparison(self, tx, left, right):
         def _unimplemented():
             unimplemented(f"comparison {typestr(left)} {op} {typestr(right)}")
 
+        if op.__name__ == "is_":
+            # If the two objects are of different type, we can safely return False
+            if type(left) is not type(right):
```

Comment:
You need to guard on the types of left/right to do this short circuit. It may not be worth it.

Reply:
Actually no this is fine

Reply:
Actually, it may not be fine.

1. Compile 1 - Enum and UserObjectVariable - so return false
2. Compile 2 - we pass in the same object. But no guards. Still returns False.

Reply:
Actually, that's too strict. So let's give up and not do this

Reply:
How about we check if both objects already have type match, and then do this? That might still be fine.

Reply:
We will restrict this PR to Tensor match